inference-engine/samples/hello_nv12_input_classification/main.cpp

```diff
@@ -49,29 +58,82 @@ std::pair<size_t, size_t> parseImageSize(const std::string& size_string) {
     return {width, height};
 }
 
+// Comparing to samples/args_helper.hpp, this version filters files by ".yuv" extension
 /**
- * \brief Read image data from file
- * @return buffer containing the image data
- */
-std::unique_ptr<unsigned char[]> readImageDataFromFile(const std::string& image_path, size_t size) {
-    std::ifstream file(image_path, std::ios_base::ate | std::ios_base::binary);
-    if (!file.good() || !file.is_open()) {
-        std::stringstream err;
-        err << "Cannot access input image file. File path: " << image_path;
-        throw std::runtime_error(err.str());
+* @brief This function checks input args and existence of specified files in a given folder
+* @param path path to a file to be checked for existence
+* @return files updated vector of verified input files
+*/
+std::vector<std::string> readInputFileNames(const std::string& path) {
+    struct stat sb;
+    if (stat(path.c_str(), &sb) != 0) {
+        slog::warn << "File " << path << " cannot be opened!" << slog::endl;
+        return {};
     }
 
-    const size_t file_size = file.tellg();
-    if (file_size < size) {
-        std::stringstream err;
-        err << "Invalid read size provided. File size: " << file_size << ", to read: " << size;
-        throw std::runtime_error(err.str());
+    std::vector<std::string> files;
+
+    if (S_ISDIR(sb.st_mode)) {
+        DIR* dp;
+        dp = opendir(path.c_str());
```

Comment:
Nitpick: why not `DIR* dp = opendir(path.c_str());`?

Reply:
Thank you.
It is artifact of copying another IE code: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/samples/common/samples/args_helper.hpp#L38